src/log/log_sys.c

```diff
@@ -9,11 +9,54 @@
 #include "wt_internal.h"
 
 /*
- * __wt_log_system_record --
+ * __wt_log_system_backup_id --
+ *     Write a system log record for the incremental backup IDs.
+ */
+int
+__wt_log_system_backup_id(WT_SESSION_IMPL *session)
+{
+    WT_BLKINCR *blk;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_ITEM(logrec);
+    WT_DECL_RET;
+    size_t recsize;
+    uint32_t i, rectype;
+    const char *fmt;
+
+    conn = S2C(session);
+    if (!F_ISSET(conn, WT_CONN_INCR_BACKUP))
+        return (0);
+
+    /* Set up the system log record itself. */
+    rectype = WT_LOGREC_SYSTEM;
+    fmt = WT_UNCHECKED_STRING(I);
+    WT_ERR(__wt_struct_size(session, &recsize, fmt, rectype));
+    WT_ERR(__wt_logrec_alloc(session, recsize, &logrec));
+    WT_ERR(
+      __wt_struct_pack(session, (uint8_t *)logrec->data + logrec->size, recsize, fmt, rectype));
+    logrec->size += recsize;
+
+    /*
+     * Now set up the log operation component. The pack function will grow the log record buffer as
+     * necessary.
+     */
+    for (i = 0; i < WT_BLKINCR_MAX; ++i) {
+        blk = &conn->incr_backups[i];
+        if (F_ISSET(blk, WT_BLKINCR_VALID))
```

Comment:
As this code is logging only existing backup IDs, does it also handle the case when a valid backup ID is removed (and is not replaced by a new backup ID)? For example, this would be the case of using `force_stop`.